source_files/gpu.c

```diff
@@ -35,22 +42,20 @@
 #include "gpu.h"
 
 #define CUDA_ERROR_CHECK
-#define CudaSafeCall( err ) __cudaSafeCall( err, __FILE__, __LINE__ )
-#define CudaCheckError() __cudaCheckError( __FILE__, __LINE__ )
+#define cuda_safe_call( cuerr ) __cuda_safe_call( cuerr, __FILE__, __LINE__ )
 
-static void *A,*B;
 static unsigned int size,useDouble,useDevice,verbose; //matrixsize,switch for double precision and numbers of GPUs to use.
 static gpustruct * gpuvar=NULL;
 
 //CUDA Error Checking
-inline void __cudaSafeCall( cudaError_t err, const char *file, const int line )
+inline void __cuda_safe_call( cudaError_t cuerr, const char *file, const int line )
 {
 #ifdef CUDA_ERROR_CHECK
-    if ( cudaSuccess != err && err != 1) //we ignore the error code 1, since we don't need a __global__ kernel function
+    if ( cudaSuccess != cuerr && cuerr != 1) //we ignore the error code 1, since we don't need a __global__ kernel function
```

Comment:
change order to  if (cuerr != cudaSuccess)
Please check whether we can ignore cuerr != 1. I'd doubt it